guides/md/keras_hub/semantic_segmentation_deeplab_v3.md

```diff
@@ -0,0 +1,11901 @@
+This guide can be run with any backend (Tensorflow, JAX, PyTorch).
+
+
+```python
+import os
+
+os.environ["KERAS_BACKEND"] = "jax"
+import keras
+from keras import ops
+import keras_hub
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+```
+
+---
+## Perform semantic segmentation with a pretrained DeepLabv3+ model
+
+The highest level API in the KerasHub semantic segmentation API is the
+`keras_hub.models` API. This API includes fully pretrained semantic segmentation
+models, such as `keras_hub.models.DeepLabV3ImageSegmenter`.
+
+Let's get started by constructing a DeepLabv3 pretrained on the Pascal VOC
+dataset.
+Also, define the preprocessing function for the model to preprocess images and
+labels.
+**Note:** By default `from_preset()` method in KerasHub loads the pretrained
+task weights with all the classes, 21 classes in this case.
+
+
+```python
+model = keras_hub.models.DeepLabV3ImageSegmenter.from_preset(
+    "deeplab_v3_plus_resnet50_pascalvoc"
+)
+
+image_converter = keras_hub.layers.DeepLabV3ImageConverter(
+    image_size=(512, 512),
+    interpolation="bilinear",
+)
+preprocessor = keras_hub.models.DeepLabV3ImageSegmenterPreprocessor(image_converter)
+```
+
+<div class="k-default-codeblock">
+```
+/home/sachinprasad/projects/env/lib/python3.11/site-packages/keras/src/saving/saving_lib.py:719: UserWarning: Skipping variable loading for optimizer 'adam', because it has 2 variables whereas the saved optimizer has 185 variables. 
+  saveable.load_own_variables(weights_store.get(inner_path))
+
+```
+</div>
+Let us visualize the results of this pretrained model
+
+
+```python
+filepath = keras.utils.get_file(
+    origin="https://storage.googleapis.com/keras-cv/pictures/dog.jpeg"
+)
+image = keras.utils.load_img(filepath)
+image = keras.utils.img_to_array(image)
+
+image = preprocessor(image)
+image = keras.ops.expand_dims(image, axis=0)
+preds = ops.expand_dims(ops.argmax(model(image), axis=-1), axis=-1)
+
+
+def plot_segmentation(original_image, predicted_mask):
+    original_image = np.squeeze(original_image, axis=0)
+    original_image = np.clip(original_image / 255.0, 0, 1)
+    predicted_mask = np.squeeze(predicted_mask, axis=0)
+    plt.figure(figsize=(5, 5))
+
+    plt.subplot(1, 2, 1)
+    plt.imshow(original_image)
+    plt.axis("off")
+
+    plt.subplot(1, 2, 2)
+    plt.imshow(predicted_mask, cmap="gray")
+    plt.axis("off")
+
+    plt.tight_layout()
+    plt.show()
+
+
+plot_segmentation(image, preds)
+```
+
+<div class="k-default-codeblock">
+```
+/home/sachinprasad/projects/env/lib/python3.11/site-packages/keras/src/models/functional.py:225: UserWarning: The structure of `inputs` doesn't match the expected structure: ['keras_tensor']. Received: the structure of inputs=*
+  warnings.warn(
+
+```
+</div>
+    
+![png](/home/sachinprasad/projects/keras-io/guides/img/semantic_segmentation_deeplab_v3/semantic_segmentation_deeplab_v3_9_1.png)
```

Comment:
there is some issue with autogen. The file paths need to updated to be relative here

Reply:
Made changes manually, could you please check.